docs/guides/auth/index.rst

```diff
@@ -269,8 +300,94 @@ To enable any of the others, change ``AzureOAuthProvider`` in the example above
 to one of the other providers:
 
 - ``AppleOAuthProvider``
+- ``DiscordOAuthProvider``
 - ``GitHubOAuthProvider``
 - ``GoogleOAuthProvider``
+- ``SlackOAuthProvider``
+
+
+Magic link
+----------
+
+Magic link offers only one setting: ``token_time_to_live``. This determines how
+long after sending the magic link is valid.
+
+You will also need to configure SMTP so that the magic link emails can be sent.
+For local testing, you can use the same method used for SMTP previously for
+:ref:`the email and password provider
+<ref_guide_auth_overview_email_password>`.
```

Comment:
Went with a slightly different wording here. I want to make it explicit that the _reason_ you need to configure SMTP is because you'll need it to send the emails.

> Since magic links rely on email, you must also configure SMTP.